pkg/esgcet/mk_dataset.py

```diff
@@ -259,139 +256,58 @@ def get_file(self, mapdata, fn_trid):
 
         return ret
 
-    def get_scanfile_dict(self, scandata):
-        ret = {}
-        for key in scandata:
-            rec = scandata[key]
-            ret[rec['name']] = rec
-        return ret
-
-    def update_metadata(self, record, scanobj):
-        if "variables" in scanobj:
-            if self.variable_name in record:
-
-                vid = record[self.variable_name]
-                try:
-                    if vid in scanobj["variables"]:
-                        var_rec = scanobj["variables"][vid]
-                        if "long_name" in var_rec.keys():
-                            record["variable_long_name"] = var_rec["long_name"]
+    def set_variables(self, record, scanobj):
+        variables = self.get_variables(scanobj)
+        # use the correct facet id string to get the variable if pre-specified in the record
+        vid = record[self.variable_name]
+        if vid in variables:
+            var_rec = variables[vid]
+            if "long_name" in var_rec:
+                record["variable_long_name"] = var_rec["long_name"]
+            elif "info" in var_rec:
+                record["variable_long_name"] = var_rec["info"]
```

Comment:
```suggestion
                record["variable_info"] = var_rec["info"]
```

Reply:
Or was this intentional?

Reply:
Also, this section could be written in a more easily extensible loop.
```
        if vid in variables:
            var_rec = variables[vid]
            record[self.variable_name] = vid
            if self.variable_name == "variable_id":
                record["variable"] = vid
            for attr in ["long_name","info","standard_name","units"]:
                record[f"variable_{attr}"] = var_rec[attr] 
```

Reply:
`variable_long_name` is the standard ESGF search field we need to populate, also `cf_standard_name`.  So we are stuck with particularly formatted keys, also some files use `long_name` and others use `info` instead.  This is one area where ESGF adopted a standard where there may not have been anything so strict.

Reply:
Of course, there's no documentation (too often I've been in a hurry) so that would improve these things...